User Management, VNet CIDR, and API Management Updates (#83)
bcgov/public-cloud-techdocs: docs/azure/design-build-deploy/networking.md

Last updated: **{{ git_revision_date_localized }}**

The following sections describe the networking components within the Azure Landing Zone, including the Virtual Network (VNet), spoke-to-spoke connectivity, and internet connectivity.

## Virtual network (VNet)

Within each Project Set deployed in the Azure Landing Zone, a [Virtual Network (VNet)](https://learn.microsoft.com/en-us/azure/virtual-network/virtual-networks-overview) is created to provide network isolation and security for the resources deployed within it. This VNet is the foundation for all network connectivity within the Azure Landing Zone.

!!! danger "VNet CIDR Changes"
    The CIDR range for the VNet is defined at the time of deployment, and cannot be changed after deployment. By default, each Project Set is provided with a `/24` CIDR range. If you require a change to the  CIDR range, please [submit a Service Request](https://citz-do.atlassian.net/servicedesk/customer/portal/3) to the Public Cloud team.

This VNet is connected with the central hub (vWAN), and receives default routes to direct all traffic (ie. Internet and private) through the firewall located in the central hub.

> There are no subnets that are pre-created within the VNet. Each team is responsible for creating their own subnets based on their requirements. Subnets should be created within the VNet to segment resources based on their function or security requirements.

Reviewer: ```suggestion
There aren't subnets that are pre-created within the VNet. Each team is responsible for creating their own subnets based on their requirements. Subnets should be created within the VNet to segment resources based on their function or security requirements.
```

For more complex applications, an [Azure Application Gateway](https://learn.microsoft.com/en-us/azure/application-gateway/overview) is the preferred method for exposing your application to the Internet. It provides a web traffic (OSI layer 7) load balancer that enables you to manage traffic to your web applications.

To adhere to security best practices, the Application Gateway should also be configured with a [Web Application Firewall (WAF)](https://learn.microsoft.com/en-us/azure/application-gateway/features#web-application-firewall) to protect your applications from common exploits and vulnerabilities.

## Protected network resources

In order to maintain the security of the Azure Landing Zone, there are certain network resources that are protected and cannot be modified by teams, and other network resources that cannot be created in the Landing Zone. These include:

* Modifying the Virtual Network (VNet) DNS settings
  * This is required so that all traffic is routed through the central firewall, for compliance requirements
* Creating Express Route circuits, VPN Sites, VPN/NAT/Local Gateways, or Route Tables

>   * This is so that traffic is not bypassing the central firewall

Reviewer: ```suggestion
  * This prevents traffic from bypassing the central firewall
```

* Creating Virtual Networks

>   * This is to avoid overlapping IP address ranges that may be in use by other Project Sets

Reviewer: ```suggestion
  * This prevents overlapping IP address ranges that may be in use by other Project Sets
```

* Creating Virtual Network peering with other VNets

>   * This is required to ensure spoke-to-spoke traffic is managed centrally through the firewall

Reviewer: ```suggestion
  * This ensures spoke-to-spoke traffic is centrally managed through the firewall
```

> * Deleting the `setbypolicy` Diagnostics Settings

Reviewer: ```suggestion
* Deleting the `setbypolicy` diagnostics settings
```

>   * You can create your own Diagnostics Settings for your resources, but you cannot delete the default one

Reviewer: ```suggestion
  * You can create your own diagnostics settings for your resources, but you can't delete the default one
```
